sklearn_pandas/dataframe_mapper.py

```diff
@@ -171,27 +174,33 @@ def fit(self, X, y=None):
                       self._get_col_subset(X, self._unselected_columns(X)), y)
         return self
 
-    def get_names(self, c, t, x):
+    def get_names(self, columns, transformer, x, alias=None):
         """
         Return verbose names for the transformed columns.
 
-        c       name (or list of names) of the original column(s)
-        t       transformer
-        x       transformed columns (numpy.ndarray)
+        columns       name (or list of names) of the original column(s)
+        transformer   transformer
+        x             transformed columns (numpy.ndarray)
+        alias         base name to use for the selected columns
         """
-        if isinstance(c, list):
-            c = '_'.join(c)
+        if alias is not None:
+            name = alias
+        elif isinstance(columns, list):
+            name = '_'.join(columns)
+        else:
+            name = columns
         num_cols = x.shape[1] if len(x.shape) > 1 else 1
         if num_cols > 1:
-            # If there are as many columns as classes,
+            # If there are as many columns as classes in the transformer,
             # infer column names from classes names.
-            if hasattr(t, 'classes_') and (len(t.classes_) == num_cols):
-                return [c + '_' + str(o) for o in t.classes_]
+            if hasattr(transformer, 'classes_') and \
```

Comment:
I agree we can do that, but in a different PR. This one is to allow specifying a custom name for the extracted columns. Can you submit a PR with that?